Packet++/header/TpktLayer.h

```diff
@@ -0,0 +1,144 @@
+#ifndef PACKETPP_TPKT_LAYER
+#define PACKETPP_TPKT_LAYER
+
+#include "EthLayer.h"
+#include "Layer.h"
+
+/// @file
+
+/**
+ * \namespace pcpp
+ * \brief The main namespace for the PcapPlusPlus lib
+ */
+namespace pcpp
+{
+
+	/**
+	 * @struct tpkthdr
+	 * Represents a TPKT protocol header
+	 */
+#pragma pack(push, 1)
+	struct tpkthdr
+	{
+		/** message version */
+		uint8_t version;
+		/** message reserved */
+		uint8_t reserved;
+		/** message length */
+		uint16_t length;
+	};
+#pragma pack(pop)
+
+	/**
+	 * @class TpktLayer
+	 * Represents a TPKT (Transport Service on top of the TCP) protocol layer
+	 */
+	class TpktLayer : public Layer
+	{
+	  public:
+		/**
+		 * A constructor that creates the layer from an existing packet raw data
+		 * @param[in] data A pointer to the raw data (will be casted to @ref tpkthdr)
+		 * @param[in] dataLen Size of the data in bytes
+		 * @param[in] prevLayer A pointer to the previous layer
+		 * @param[in] packet A pointer to the Packet instance where layer will be stored in
+		 */
+		TpktLayer(uint8_t *data, size_t dataLen, Layer *prevLayer, Packet *packet)
+			: Layer(data, dataLen, prevLayer, packet)
+		{
+			m_Protocol = TPKT;
+		}
+
+		/**
+		 * A constructor that allocates a new TPKT header
+		 * @param[in] version the value of the version
+		 * @param[in] length the value of the length
+		 */
+		TpktLayer(uint8_t version, uint16_t length)
+		{
+			m_DataLen = sizeof(tpkthdr);
+			m_Data = new uint8_t[m_DataLen];
+			memset(m_Data, 0, m_DataLen);
+			tpkthdr *tpktHdr = getTpktHeader();
+			tpktHdr->version = version;
+			tpktHdr->reserved = 0;
+			tpktHdr->length = htobe16(length);
+			m_Protocol = TPKT;
```

Comment:
We should move the implementation of the constructor to `TpktLayer.cpp`.

[CI fails](https://github.com/seladb/PcapPlusPlus/actions/runs/4569020944/jobs/8067329483?pr=1098) because `#include "EndianPortable.h"` that is required for using `htobe16()` is only defined in [`TpktLayer.cpp`](https://github.com/wivien19/PcapPlusPlus/blob/79ece62965a833f2c5b4eace3768e2a0f0435d51/Packet%2B%2B/src/TpktLayer.cpp#L2)